python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -367,17 +367,34 @@ def computeCorrection(self, kappa, svar, tvar, preConvArr=None):
         The maximum correction factor converges to `sqrt(tvar/svar)` towards high frequencies.
         This should be a plausible value.
         """
+        kSum = np.sum(kappa)
         kappa = self.padCenterOriginArray(kappa, self.freqSpaceShape)
         kft = np.fft.fft2(kappa)
-        kft2 = np.real(np.conj(kft) * kft)
-        if preConvArr is None:
-            denom = svar + tvar * kft2
-        else:
+        kftAbsSq = np.real(np.conj(kft) * kft)
+        # If there is no pre-convolution kernel, use placeholder scalars
+        preSum = 1.
+        preAbsSq = 1.
+        if preConvArr is not None:
+            preSum = np.sum(preConvArr)
             preConvArr = self.padCenterOriginArray(preConvArr, self.freqSpaceShape)
-            mk = np.fft.fft2(preConvArr)
-            mk2 = np.real(np.conj(mk) * mk)
-            denom = svar * mk2 + tvar * kft2
-        kft = np.sqrt((svar + tvar) / denom)
+            preK = np.fft.fft2(preConvArr)
+            preAbsSq = np.real(np.conj(preK)*preK)
+
+        denom = svar * preAbsSq + tvar * kftAbsSq
+        # Division by zero protection, though we don't expect to hit it
+        # (rather we'll have numerical noise)
+        tiny = np.finfo(kftAbsSq.dtype).tiny * 1000.
+        flt = denom < tiny
+        if np.sum(flt) > 0:
```

Comment:
As I asked during standup: would it be possible to test for this bit of code int he unittests?

Another option, since on the ticket you mention this should be a rare (or possibly impossible) occurrence, would it make sense to add a warning here for when the `if` statement is tripped?

Reply:
Added a warning for this case.